core/matrix/coo.cpp

```diff
@@ -143,6 +146,123 @@ void Coo<ValueType, IndexType>::move_to(
+            if (d < 0x7d) {
+                p++;
+            } else if (d < 0xffff) {
+                p += 3;
+            } else {
+                p += 5;
+            }
+            c = col_idxs[k];
+        }
+                                offsets[b+1] = p;
+    }
+    return p;
+}
+*/
+
+
+template <typename ValueType, typename IndexType>
+void Coo<ValueType, IndexType>::convert_to(
+    Bccoo<ValueType, IndexType>* result) const  // GKO_NOT_IMPLEMENTED;
+/* */
+{
+    auto exec = this->get_executor();
+    auto num_stored_elements = this->get_num_stored_elements();
+
+    // const auto block_size = 1024;
+    const size_type block_size = 1024;
+    //		const auto block_size = Bccoo<ValueType,
+    // IndexType>::compute_block_size(
+    // result->get_executor(), this.size(), num_stored_elements);
+    const auto num_blocks = ceildiv(num_stored_elements, block_size);
+
+    Array<IndexType> rows(exec, num_blocks);
+    Array<IndexType> offsets(exec, num_blocks + 1);
+
+    size_type mem_size{};
+    if (exec == exec->get_master()) {
+        exec->run(coo::make_mem_size_bccoo(this, rows.get_data(),
+                                           offsets.get_data(), num_blocks,
+                                           block_size, &mem_size));
+        /*
+                exec->run(coo::make_mem_size_bccoo(
+                    //			mem_size_bccoo(
+                    this->get_const_row_idxs(), this->get_const_col_idxs(),
+                    //				this->get_size()[0], rows,
+           offsets,
+                    // block_size, &mem_size);
+                    this->get_size()[0], rows.get_data(), offsets.get_data(),
+                    num_stored_elements, num_blocks, block_size, &mem_size));
+        */
+    } else {
+        auto host_coo = clone(exec->get_master(), this);
+        exec->run(coo::make_mem_size_bccoo(host_coo.get(), rows.get_data(),
+                                           offsets.get_data(), num_blocks,
+                                           block_size, &mem_size));
+        /*
+                exec->run(coo::make_mem_size_bccoo(
+                    host_coo->get_const_row_idxs(),
+           host_coo->get_const_col_idxs(), host_coo->get_size()[0],
+           rows.get_data(), offsets.get_data(), num_stored_elements, num_blocks,
+           block_size, &mem_size));
+        */
+    }
+
+    Array<uint8> data(exec, mem_size);
+
+    auto tmp = Bccoo<ValueType, IndexType>::create(
+        //        exec, this->get_size(), data.get_data(), offsets, rows,
+        //        num_stored_elements,
+        exec, this->get_size(), std::move(data), std::move(offsets),
+        std::move(rows), num_stored_elements, block_size);
+
+    exec->run(coo::make_convert_to_bccoo(this, tmp.get()));
+    tmp->move_to(result);
+}
+/* */
+
+template <typename ValueType, typename IndexType>
+void Coo<ValueType, IndexType>::move_to(Bccoo<ValueType, IndexType>* result)
+    GKO_NOT_IMPLEMENTED;
```

Comment:
Why is this not implemented? Since you don't seem to re-use any existing array in the `convert_to` method above, you could simply call the `convert_to` method from here:
```suggestion
template <typename ValueType, typename IndexType>
void Coo<ValueType, IndexType>::move_to(Bccoo<ValueType, IndexType>* result) {
    this->convert_to(result);
}
```
Also, commented code below needs to be removed.

Reply:
Done